examples/step-57/step-57.cc

```diff
@@ -0,0 +1,879 @@
+  {
+    std::ostringstream filename;
+    filename << (1/viscosity) << "-line-" << refinement << ".txt";
+
+    std::ofstream f (filename.str().c_str());
+    f << "# y u_x u_y" << std::endl;
+
+    Point<dim> p;
+    p(0)= 0.5;
+    p(1)= 0.5;
+
+    f << std::scientific;
+
+    for (unsigned int i=0; i<=100; ++i)
+      {
+        p(dim-1) = i/100.0;
+
+        Vector<double> tmp_vector(dim+1);
+        VectorTools::point_value(dof_handler, present_solution, p, tmp_vector);
+        f << p(dim-1);
+
+        for (int j=0; j<dim; j++)
+          f << " " << tmp_vector(j);
+        f << std::endl;
+      }
+  }
+
+
+  // @sect4{StationaryNavierStokes::run}
+  // This is the last step of this program. In this part, we generate the grid and run
+  // the other functions respectively. The max refinement can be set by the argument.
+  template <int dim>
+  void StationaryNavierStokes<dim>::run(const unsigned int refinement)
+  {
+
+    GridGenerator::hyper_cube(triangulation);
+    triangulation.refine_global(5);
+
+    const double Re =  1.0/viscosity;
+
+    // If the viscosity is smaller than 1/1000, we have to first search for an
+    // initial guess via a continuation method. What we should notice is the
+    // search is always on the initial mesh, that is the $8 \times 8$ mesh in
+    // this program. After that, we just do the same as we did when viscosity
+    // is larger than 1/1000: run Newton's iteration, refine the mesh,
+    // transfer solutions, and repeat.
+    if (Re > 1000.0)
+      {
+        std::cout << "       Searching for initial guess ... " << std::endl;
+        const double step_size = 2000.0;
+        search_initial_guess(step_size);
+        std::cout << "*****************************************" << std::endl
+                  << "       Initial guess obtained            " << std::endl
+                  << "                  *                      " << std::endl
+                  << "                  *                      " << std::endl
+                  << "                  *                      " << std::endl
+                  << "                  *                      " << std::endl
+                  << "*****************************************" << std::endl;
+
+        std::cout << "       Computing solution with target Re = " << Re << std::endl;
+        viscosity = 1.0/Re;
+        newton_iteration(1e-12, 50, refinement, false, true);
+      }
+    else
+      {
+        // When the viscosity is larger than 1/1000, the solution to Stokes
+        // equations is good enough as an initial guess. If so, we do not need
+        // to search for the initial guess using a continuation
+        // method. Newton's iteration can be started directly.
+
+        newton_iteration(1e-12, 50, refinement, true, true);
+      }
+  }
+}
+
+int main()
+{
+  using namespace dealii;
+  using namespace Step57;
+
+  deallog.depth_console(0);
+
+  StationaryNavierStokes<2> flow(1 /*degree*/);
+  flow.run(4);
```

Comment:
please use the same template as in all other tutorials starting at step-5 that includes the try-catch block.